crates/engine/tree/src/tree/root.rs

```diff
@@ -347,6 +347,22 @@ where
     /// Spawns a new multiproof calculation or enqueues it for later if
     /// `max_concurrent` are already inflight.
     fn spawn_or_queue(&mut self, input: MultiproofInput<Factory>) {
+        // If there are no proof targets, we can just send an empty multiproof back immediately
+        if input.proof_targets.is_empty() {
+            let _ = input.state_root_message_sender.send(StateRootMessage::ProofCalculated(
+                Box::new(ProofCalculated {
+                    sequence_number: input.proof_sequence_number,
+                    update: SparseTrieUpdate {
+                        state: input.hashed_state_update,
+                        targets: input.proof_targets,
+                        multiproof: MultiProof::default(),
+                    },
+                    elapsed: Duration::ZERO,
+                }),
+            ));
+            return
```

Comment:
should we just not generate new `proof_sequence_number` and not send anything on empty targets?

Reply:
it will break the state machine a bit, because we collect all proofs and state updates on every `StateRootMessage::ProofCalculated` message, and then send an aggregate to the sparse trie

Reply:
the reason we need to send `StateRootMessage::ProofCalculated` here is because we need to send `input.hashed_state_update` to the sparse trie somehow